### PR TITLE
Laravel 5.3 Shift

### PR DESCRIPTION
**Pull request by laravel-shift**
This pull request includes the changes for upgrading to Laravel 5.3. Feel free to commit any additional changes to the `laravel-5.3-shift` branch.

**Before merging**, you should:
- Checkout the `laravel-5.3-shift` branch
- Review **all** pull request comments for additional changes
- Run `composer update` (if the pre-commands fail, add `--no-scripts`)
- Thoroughly test your application

If you would like more help with your upgrade, check out the new [human services from Laravel Shift](https://laravelshift.com/human-services).

### CODE REVIEW

**Comment by laravel-shift** [comment]
ℹ Laravel 5.3 updated the  `app/Jobs` folder by default. It is created for you when you make your first *Job* using `artisan`. I did not remove yours as it appears to differ from the default. However, if your app does not use *Jobs*, you can remove this folder manually to keep your app streamlined.

**Comment by laravel-shift** [comment]
ℹ Laravel 5.3 no longer includes the `app/Events` folder by default. It is created for you when you make your first *Event* using `artisan`. I did not remove yours as it appears to differ from the default. However, if your app does not use *Events*, you can remove this folder manually to keep your app streamlined.

**Comment by laravel-shift** [comment]
❌ I was not able to update your `app/Providers/AuthServiceProvider.php`. You will need to compare yours against the [5.3 version](https://github.com/laravel-shift/laravel-5.3/blob/master/app/Providers/AuthServiceProvider.php) and merge the new changes.

**Comment by laravel-shift** [comment]
❌ I was not able to update your `app/Providers/EventServiceProvider.php`. You will need to compare yours against the [5.3 version](https://github.com/laravel-shift/laravel-5.3/blob/master/app/Providers/EventServiceProvider.php) and merge the new changes.

**Comment by laravel-shift** [comment]
❌ I was not able to update your `app/Providers/RouteServiceProvider.php`. You will need to compare yours against the [5.3 version](https://github.com/laravel-shift/laravel-5.3/blob/master/app/Providers/RouteServiceProvider.php) and merge the new changes.

**Comment by laravel-shift** [comment]
⚠ Laravel 5.3 now includes a top-level `routes` folder. In addition, it separates routes into *API*, *console*, and *web* files.I have included a default `routes/api.php` and `routes/console.php` as well as migrated your `app/Http/routes.php` to `routes/web.php`. You should review your routes to see if they should be separated into one of the new route files.

**Comment by laravel-shift** [comment]
⚠ The default behavior of `Route::resource()` changed in Laravel 5.3. All resource route parameters are now singularized by default and URL prefixes no longer affect the route names.You should review your resource routes and verify this singularized naming will not affect your parameters (or if you were doing this manually, you can remove the `parameter`). In addition, verify you are no longer appending the URI prefix to the route name.

Found `Route::resource()` in:

- [ ] routes/web.php

**Comment by laravel-shift** [comment]
❌ I could not upgrade the following configuration files since they differed from the 5.2 defaults. You need to compare these configuration files against the [Laravel 5.3 configuration files](https://github.com/laravel-shift/laravel-5.3/blob/master/config) and merge any changes:

- [ ] config/auth.php
- [ ] config/cache.php
- [ ] config/database.php
- [ ] config/services.php
- [ ] config/session.php

**Comment by laravel-shift** [comment]
❌ I could not upgrade the following project files since they differed from the 5.2 defaults. You need to compare these project files against the [default Laravel 5.3 versions](https://github.com/laravel-shift/laravel-5.3/) and merge any changes:

- [ ] gulpfile.js
- [ ] package.json

**Comment by laravel-shift** [comment]
❌ Laravel 5.3 `where` performs a "loose" comparison. In addition, the *strict* parameter was removed. If you need to perform a "strict" comparison, you can use `whereStrict` instead.

You should review the following usages of `where` to ensure you do not require "strict" compaison:

- [ ] app/Models/Star.php
- [ ] app/Models/Tag.php
- [ ] app/Http/Controllers/AuthController.php
- [ ] app/Http/Controllers/StarController.php
- [ ] app/Http/Controllers/TagController.php

**Comment by laravel-shift** [comment]
❌ In Laravel 5.3, Eloquent scopes now respect the leading boolean of scope constraints. If you are starting your scope with an `orWhere` constraint it will no longer be converted to a normal `where`.

You should review the following usages of `orWhere` to ensure you're not starting your constraints with it:

- [ ] app/Console/Commands/AddTagSlugs.php

**Comment by laravel-shift** [comment]
❌ The Query Builder returns collections, instead of plain arrays, in Laravel 5.3. You will need to upgrade your code to use collections or chain the `all()` method onto your query to return a plain array.

You should review the following usages of Query Builder:

- [ ] app/Models/Star.php

**Comment by laravel-shift** [comment]
ℹ If you are using *Queues*, you no longer need to specify the `--daemon` option when calling the `queue:work`. This is assumed in Laravel 5.3. Use the `--once` option to process a single job.

If you are using the `database` driver, you will need to update your `jobs` and `failed_jobs` table. Review the *Queue* section of the [Laravel 5.3 Upgrade Guide](https://laravel.com/docs/5.3/upgrade#upgrade-5.3.0) for more details and feel free to use the [migrations from Laravel Shift](https://github.com/laravel-shift/laravel-5.3/tree/master/database/migrations).

**Comment by laravel-shift** [comment]
ℹ Laravel 5.3 updated several of the default assets as well as adding support for [Vue.js](https://vuejs.org). If you are using Laravel to [compile assets](https://laravel.com/docs/5.3/elixir) you may want to review the latest versions of [`resources/assets/js/app.js`](https://github.com/laravel-shift/laravel-5.3/blob/master/resources/assets/js/app.js) and [`resources/assets/sass/app.scss`](https://github.com/laravel-shift/laravel-5.3/blob/master/resources/assets/sass/app.scss) and update your project accordingly.

**Comment by laravel-shift** [comment]
⚠ Laravel 5.3 added configuration for your application name in `config/app.php`. Be sure to set this value as it is used in notifications, like the password reset email.

**Comment by syropian** [comment]
@alexbowers Do you think you could handle any additional changes that need to be made to get this PR merge ready? You probably have a better idea of what to do for some the ❌'s than me 😄

**Comment by alexbowers** [comment]
Sure, probably sunday is when I will get the chance to look at this.

**Comment by syropian** [comment]
Great, thanks! I'm away this weekend, so I won't be able to do any code changes.

File: app/Http/Kernel.php [hunk 1]
```diff
@@ -45,8 +47,9 @@ class Kernel extends HttpKernel
      * @var array
      */
     protected $routeMiddleware = [
-        'auth' => \Astral\Http\Middleware\Authenticate::class,
```
Comment (alexbowers) [inline app/Http/Kernel.php]: This needs reviewing. Did we do anything special here?

**Comment by syropian** [comment]
@alexbowers I think we should prioritize getting this done. Would be nice to have 5.3 for any future server-side changes we do for star loading performance.

**Comment by alexbowers** [comment]
I'll hopefully be able to make a start on this this evening. I'll see how far I manage to get :)

**Comment by alexbowers** [comment]
@syropian I think this is okay, however the CSRF has been disabled, Im not sure how it was previously; if there was something special done can you look at it.

**Comment by syropian** [comment]
I don't think we were using CSRF tokens at any point (should we be?)

**Comment by alexbowers** [comment]
In reality, yes.

But, its good to know that we dont currently, means i haven't made the
system any less secure.

**Comment by alexbowers** [comment]
All this requires is your approval

**Comment by syropian** [comment]
@alexbowers Since a JWT token is sent as an authorization header with almost every request, and we aren't using cookies to store any session data, I think it negates the need for CSRF.

**Comment by syropian** [comment]
This looks good btw, 👍 :shipit:

**Comment by alexbowers** [comment]
Can you do some manual testing on features?

Or merge and see what breaks. Whichever way :)

**Comment by syropian** [comment]
I'm going to merge this, and test on master. Won't deploy until I do some manual testing. We can create additional branches for any fixes that arise.

**Pull request merged**